src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +30,43 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] addressValues=removePrefix(trimmedAddress, ADDRESS_PREFIX).split(",", 4);
```

Comment:
what does 4 mean? why only 4? why not 5 , 2 or 100? 
I hope you get what I mean :-)